File: src/backend/shared/credential.py
from azure.identity.aio import DefaultAzureCredential

from config import settings


def create_credential() -> DefaultAzureCredential:
    """Create an async ``DefaultAzureCredential``.

    When ``AZURE_CLIENT_ID`` is set (i.e. running in Azure with a
    user-assigned managed identity), the credential is scoped to that
    identity.  When the value is empty (local development), plain
    ``DefaultAzureCredential()`` is returned so the Azure CLI / VS Code
    login flow works automatically.
    """
    if settings.azure_client_id:
        return DefaultAzureCredential(managed_identity_client_id=settings.azure_client_id)
    return DefaultAzureCredential()

File: src/worker-analysis-fsharp/Credential.fs
/// Azure credential factory — mirrors Python shared/credential.py.
module IntegrisightWorkerAnalysis.Credential

open Azure.Identity

/// Create a DefaultAzureCredential.
/// When AZURE_CLIENT_ID is set (user-assigned managed identity), scope to that identity.
let createCredential (clientId: string) : DefaultAzureCredential =
    if System.String.IsNullOrEmpty clientId then
        DefaultAzureCredential()
    else
        DefaultAzureCredential(DefaultAzureCredentialOptions(ManagedIdentityClientId = clientId))
